### PR TITLE
[TkAl] Realign DQM and Offline analysis of μμ + vertex events

### PR DESCRIPTION
**Pull request by mmusich**
#### PR description:

Title says it all. After offline investigation (mainly from @vbotta) some discrepancy was found when comparing the results from the `DiMuonVertexValidation` module (in the package `Alignment/OfflineValidation`) and `DiMuonVertexMonitor` (in the package `DQMOffline/Alignment`).
These were traced back to the fact that the DQM module (by default) selects the primary vertex closest to the di-muon vertex, while the one in `Alignment/OfflineValidation` selects the first one (presumably the hardest in the event, but not guaranteed because there is no re-sorting post-refit). This is addressed in 2e70c3b9b0a1004584c02be33e7debd107a7aa62 by offering the user the possibility to use the closest (to X→μμ decay vertex) or the first primary vertex.
The other two commits:
   * 79451cfaeb1501f149aee6d3e530a98f91acee40 updates slightly the unit test configuration for `DiMuonVertexValidation`,
   *  4839b20b27939a69ac9661f4c04be251daae280d adds some debug statement in `DiMuonVertexMonitor`, that were useful during the debugging process.

#### PR validation:

Run successfully:
   * `scram b runtests_DiMuonVertex use-ibeos`
   * `scram b runtests_testDiMuonVertexMonitor use-ibeos`

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

N/A

### CODE REVIEW

**Comment by mmusich** [comment]
@cmsbuild, please test

File: Alignment/OfflineValidation/plugins/DiMuonVertexValidation.cc [hunk 1]
```diff
@@ -323,25 +345,40 @@ void DiMuonVertexValidation::analyze(const edm::Event& iEvent, const edm::EventS
...
+    edm::LogWarning("DiMuonVertexMonitor") << "invalid vertex collection encountered Skipping event!";
+    return;
   }
 
+  reco::Vertex TheMainVertex;
+  if (!useClosestVertex_ || theClosestVertex == nullptr) {
+    TheMainVertex = *theClosestVertex;
+  } else {
+    TheMainVertex = vertexHandle.product()->front();
+  }
```
Comment (perrotta) [inline Alignment/OfflineValidation/plugins/DiMuonVertexValidation.cc:365]: Uhm, shouldn't it be the opposite?
```suggestion
  if (!useClosestVertex_ || theClosestVertex == nullptr) {
    TheMainVertex = vertexHandle.product()->front();
  } else {
    TheMainVertex = *theClosestVertex;
  }
```

**Comment by perrotta** [comment]
+alca

**Comment by tjavaid** [comment]
+1

**Comment by rappoccio** [comment]
+1

**Pull request merged**